test/autorest/generated/azurespecialsgroup/xmsclientrequestid.go

```diff
@@ -63,6 +65,13 @@ func (client *xmsClientRequestIdoperations) getHandleResponse(resp *azcore.Respo
 
 // getHandleError handles the Get error response.
 func (client *xmsClientRequestIdoperations) getHandleError(resp *azcore.Response) error {
+	if resp.Body != nil {
```

Comment:
I would invert this to keep your happy-line happy.
```go
if resp.Body == nil {
  return errors.New(resp.Status)
}
...
```

Reply:
And actually, checking for `nil` is incorrect as the `Body` field will always be non-nil according to the standard lib.  Now we could change this in the body downloader policy to set `Body` to nil if it was empty however it changes the contract given by the stdlib so I'm not in favor of that.
Maybe we should have a method on `azcore.Response` to create a generic error from the response?

Reply:
Hmmm...well I agree that we shouldn't do something that changes the contract for the stdlib. Maybe it could be worth it to have an ErrorFromResponse or PayloadError method on `azcore.Response`. Having a generic error for `azcore.Response` does sound nice. It would simplify things. However, I will say that it would add more surface area to azcore, while keeping it here would be an implementation detail of handling errors for each service.

Reply:
So are you thinking that we could add something in azcore.Response that uses the payload() method? It would be convenient.

Reply:
I was thinking about the public surface area thing too, I'm a bit torn on it, but we should probably avoid it.  I guess this means that you have to read the body and if the length is zero then return the generic error.

Reply:
Yes uses payload, just like the unmarshallers.

Reply:
Ok cool, will use the length of the body.

Reply:
Done